docs/current/sdk/go/guides/421437-work-with-host-filesystem.md

```diff
@@ -0,0 +1,283 @@
+
+The `Host` type provides information about the host's execution environment. Its `Directory()` function accepts a path and returns a reference to the corresponding host directory as a `Directory` struct. There's also a shortcut `Workdir()` function, which returns a reference to the current working directory on the host. Entries in the directory can be obtained via the `Directory.Entries()` function.
+
+The following example shows how to list the contents of the host working directory:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir("."))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir().Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+## List directory contents with filters
+
+It's possible to restrict a `Directory` to a subset of directory entries, by specifying a list of filename patterns to include or exclude.
+
+The following example shows how to obtain a reference to the host working directory containing only `*.rar` files:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+  "os"
+  "path/filepath"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  dir := os.TempDir()
+  os.WriteFile(filepath.Join(dir, "foo.txt"), []byte("1"), 0600)
+  os.WriteFile(filepath.Join(dir, "bar.txt"), []byte("2"), 0600)
+  os.WriteFile(filepath.Join(dir, "baz.rar"), []byte("3"), 0600)
+
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir(dagger.HostWorkdirOpts{
+    Include: []string{"*.rar"},
+  }).Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+The following example shows how to obtain a reference to the host working directory containing all files except `*.txt` files:
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+  "log"
+  "os"
+  "path/filepath"
+
+  "dagger.io/dagger"
+)
+
+func main() {
+  dir := os.TempDir()
+  os.WriteFile(filepath.Join(dir, "foo.txt"), []byte("1"), 0600)
+  os.WriteFile(filepath.Join(dir, "bar.txt"), []byte("2"), 0600)
+  os.WriteFile(filepath.Join(dir, "baz.rar"), []byte("3"), 0600)
+
+  ctx := context.Background()
+
+  client, err := dagger.Connect(ctx, dagger.WithWorkdir(dir))
+  if err != nil {
+    log.Fatal(err)
+  }
+  defer client.Close()
+
+  // highlight-start
+  entries, err := client.Host().Workdir(dagger.HostWorkdirOpts{
+    Exclude: []string{"*.txt"},
+  }).Entries(ctx)
+  // highlight-end
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  fmt.Println(entries)
+}
+```
+
+:::note
+The `Exclude` pattern overrides the `Include` pattern, but not vice-versa.
```

Comment:
Great to have this note since I had confused myself with this in the past. :slightly_smiling_face: